docs/signature-layout.md

```diff
@@ -0,0 +1,35 @@
+# Signature File Layout
+
+A common file layout for storing and serving signatures provides a consistent way to reference image signatures. Signatures on a filesystem or a web server shall use this common layout. Signatures stored in a REST API are not required to use this common layout.
+
+## Specification
+
+This specification relies on [RFC3986](https://tools.ietf.org/html/rfc3986), focusing on defining a [path component](https://tools.ietf.org/html/rfc3986#section-3.3) to compose a concise URI reference to a signature.
+
+**SCHEME[AUTHORITY]/PATH_PREFIX/IMAGE@MANIFEST_DIGEST/signature-INT**
+
+**Definitions**
+
+* **SCHEME**: URI scheme per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.1), e.g. **file://** or **https://**
+* **AUTHORITY**: An optional authority reference per [RFC3986](https://tools.ietf.org/html/rfc3986#section-3.2), e.g. **example.com**
+* **PATH_PREFIX**: An arbitrary base path to the image component
+* **IMAGE**: The name of the image per [v2 API](https://docs.docker.com/registry/spec/api/#/overview). This would typically take the form of registry/repository/image but is not required to have exactly three parts. There is no requirement to include a **:PORT** component but it should be included if part of the image reference.
```

Comment:
On Wed, Oct 26, 2016 at 01:46:46PM -0700, Miloslav Trmač wrote:
> > A flat directory of signatures would easily solve this issue. It has disadvantages, but it's strength is this: because a manifest is not inherently tied to a specific registry, its signatures also should not be tied to a specific registry.
>
> It wouldn’t; how flat would this directory have to be? AFAICT for your case to work, and all signatures to be automatically found for whatever registry anyone in the world decides to host, there would need to be a _single world-wide sigstore_. Who would host it? Why? Would everyone want to make their images public like that? Would they want to cede control to that host? Essentially this would recreate the “Docker Hub as a single point of failure/control” design again, with all its issues.

Having a flat blob namespace doesn't mean you have to have a single, central repository.  IPFS has a flat blob namespace, and it is federated.  You can also query multiple DHTs, or query implementations that don't use a DHT at all.  Having a flat blob namespace just means you're not baking access restrictions into your blob names.  Folks who do have blob access restrictions can always apply them to flat-namespaced blobs (“sha256:123…?  Sorry, you're not a member of any group with access to that blob”).

I'm +1 to making signatures a more integral part of the image though.  Sticking them in the image's Merkle tree makes that really obvious, although there are other ways to associate them if folks don't want them in the Merkle tree.